lib/elixir/lib/float.ex

```diff
@@ -106,11 +108,15 @@ defmodule Float do
       34.25
 
   """
-  @spec floor(float, 0..15) :: float
-  def floor(number, precision \\ 0) when is_float(number) and precision in 0..15 do
+  @spec floor(float, precision_valid_range) :: float
+  def floor(number, precision \\ 0) when is_float(number) and precision in @precision_valid_range do
     round(number, precision, :floor)
   end
 
+  def floor(number, precision) when is_float(number) do
+    raise_invalid_precision(precision)
```

Comment:
We should `raise` here instead of inside helper function.
Otherwise `floor/2` could be removed from stacktrace in case of tail call optimisation.